x-pack/plugins/security_solution/common/endpoint/index_data.ts

```diff
@@ -62,7 +65,10 @@ export async function indexHostsAndAlerts(
   alertsPerHost: number,
   fleet: boolean,
   options: TreeOptions = {},
-  DocGenerator: typeof EndpointDocGenerator = EndpointDocGenerator
+  DocGenerator: typeof EndpointDocGenerator = EndpointDocGenerator,
+  disableEndpointActionsForHost = false,
+  bothIsolatedAndNormalEndpoints = false,
+  endpointIsolated?: boolean
```

Comment:
Same here. you should be able to implement this with a custom `DocGenerator` and not introduce new arguments.